src/pages/Interferometry/index.js

```diff
@@ -1,4 +1,96 @@
-import React, { useState } from "react";
+import React, { useRef, useState } from "react";
+import useAnimationFrame from "../../useAnimation";
+
+class DoubleRect {
+  constructor(x, y, w = 50, h = 50, angle = 0, strokeRatio = 0, fillRatio = 1) {
+    this.x = x;
+    this.y = y;
+    this.w = w;
+    this.h = h;
+    this.angle = angle;
+    this.radian = this.angle * (Math.PI / 180);
+    this.strokeRatio = strokeRatio;
+    this.fillRatio = fillRatio;
+  }
+
+  draw(ctx) {
+    ctx.save();
+    ctx.translate(this.x, this.y);
+    ctx.rotate(this.radian);
+    // adjust position as a center top of filled rectangle
+    ctx.strokeRect(
+      -0.5 * this.w,
+      this.h - this.fillRatio * this.h,
+      this.w,
+      -this.strokeRatio * this.h,
+    );
+    ctx.fillRect(-0.5 * this.w, 0, this.w, this.fillRatio * this.h);
+    ctx.restore();
+  }
+}
+
+const source = new DoubleRect(50, 450, 50, 50, 0, 0, 1);
+
+const splitter1 = new DoubleRect(50, 350, 50, 10, 135, 2, 1);
+const splitter2 = new DoubleRect(250, 150, 50, 10, 135, 2, 1);
+
+const mirror1 = new DoubleRect(50, 150, 50, 10, 135, 1, 1);
+const mirror2 = new DoubleRect(250, 350, 50, 10, -45, 1, 1);
+
+function MachZehnderCanvas({ size: [width, height], refractionIndex }) {
+  const cvs = useRef(null);
+
+  useAnimationFrame(() => {
+    if (!cvs.current) return;
+    const ctx = cvs.current.getContext("2d");
+    ctx.clearRect(0, 0, width, height);
+
+    // show objects
+    source.draw(ctx);
+    splitter1.draw(ctx);
+    splitter2.draw(ctx);
+
+    mirror1.draw(ctx);
+    mirror2.draw(ctx);
+
+    // show wave propagation
+    // wave properties
+    const amp = 2; // amplitude
+    const k = 0.2; // wave number
+    const lbw = 2; // light beam width
```

Comment:
just name it as you commented like
```js
const amplitude = 2;
const waveNumberK = 0.2;
const lightBeamWidth = 2; 

```

there's no rule to restrict the length of variable names.